corehq/apps/export/utils.py

```diff
@@ -81,72 +96,106 @@ def convert_saved_export_to_export_instance(domain, saved_export, dryrun=False):
-            column_path = _convert_index_to_path_nodes(index)
-            # The old style column indexes always look like they contains no repeats,
-            # so replace that parts that could be repeats with the table path
-            column_path = table_path + column_path[len(table_path):]
-
-            system_property = _get_system_property(
-                column.index,
-                transform,
-                export_type,
-                new_table.path
-            )
-            if system_property:
-                column_path, transform = system_property
-
-            guess_types = ['ScalarItem', 'MultipleChoiceItem', 'ExportItem']
-            # Since old exports had no concept of item type, we just guess all
-            # the types and see if there are any matches.
-            for guess_type in guess_types:
-                index, new_column = new_table.get_column(
-                    column_path,
-                    guess_type,
-                    _strip_deid_transform(transform),
+            try:
+                if column.doc_type == 'StockExportColumn':
+                    # Handle stock export column separately because it's a messy edge since
+                    # it doesn't have a unique index (_id).
+                    info.append('Column is a stock column')
+
+                    index, new_column = new_table.get_column(
+                        [PathNode(name='stock')],
+                        'ExportItem',
+                        None,
+                    )
+                    if new_column:
+                        new_column.selected = True
+                        new_column.label = column.display
+                    else:
+                        raise SkipConversion('StockExportColumn not found in new export')
+                    continue
+
+                if column.transform:
+                    transform = _convert_transform(column.transform)
+                    info.append('Column has a transform {} converted to {}'.format(
+                        column.transform,
+                        transform,
+                    ))
+
+                if _is_repeat(old_table.index):
+                    index = '{table_index}.{column_index}'.format(
+                        table_index=_strip_repeat_index(old_table.index),
+                        column_index=column.index,
+                    )
+                    info.append('Column is part of a repeat: {}'. format(old_table.index))
+                column_path = _convert_index_to_path_nodes(index)
+                # The old style column indexes always look like they contains no repeats,
+                # so replace that parts that could be repeats with the table path
+                column_path = table_path + column_path[len(table_path):]
+
+                system_property = _get_system_property(
+                    column.index,
+                    transform,
+                    export_type,
+                    new_table.path
```

Comment:
w=1 is helpful for this commit: https://github.com/dimagi/commcare-hq/pull/12649/commits/d6c2ae84531f8b8dc652281992ac103a1e59aa59?w=1